### DISCUSSION TITLE
Preflight response is not successful. Status code: 405

### DISCUSSION THREAD

**Discussion by cabello** [trpc/trpc#1810] [Q&A]
Sorry for the cross-posting, anyone experiencing this issue where only one domain out of 4 domains from Vercel work with `trpc` without changing any CORS configuration? I am using trpc with `ssr: false` at the moment, not sure this would impact it.

https://github.com/vercel/community/discussions/569

**Comment by kelvindecosta** [comment]
Hey @cabello 

Ran into the same issue today.

Given that Vercel deploys the same build (code) to multiple URLs, I think that the `url` string must somehow be determined from the request origin.

Is it possible to achieve this?

Any help is appreciated!
Thanks for your time

**Reply by cabello** [reply]
I talked to someone that uses trpc and Vercel and they said that they only use a single domain because yes, that feature is always broken on the other domains.

**Comment by KATT** [comment]
Do you have multiple frontend using the same backend and each frontend is on a different domain? Then yes, you need cors in your api handler

**Reply by cabello** [reply]
The community link I posted contains a bit of extra context, Vercel deploys the same build to multiple domains, but doesn't change the build or even for each deployment so their experience is always broken if using trpc.

**Comment by jamezening** [comment]
@cabello 
Are you still experiencing this issue? I am assuming you are deploying a Next+trpc project to Vercel?

I had the same issue when I setup trpc using the example from [Usage with Next.js](https://trpc.io/docs/v9/nextjs)

```
import { withTRPC } from '@trpc/next';
import { AppType } from 'next/dist/shared/lib/utils';
import { AppRouter } from './api/trpc/[trpc]';

const MyApp: AppType = ({ Component, pageProps }) => {
  return <Component {...pageProps} />;
};

export default withTRPC<AppRouter>({
  config({ ctx }) {
    /**
     * If you want to use SSR, you need to use the server's full URL
     * @link https://trpc.io/docs/ssr
     */
    const url = process.env.VERCEL_URL
      ? `https://${process.env.VERCEL_URL}/api/trpc`
      : 'http://localhost:3000/api/trpc';

    return {
      url,
      /**
       * @link https://react-query.tanstack.com/reference/QueryClient
       */
      // queryClientConfig: { defaultOptions: { queries: { staleTime: 60 } } },
    };
  },
  /**
   * @link https://trpc.io/docs/ssr
   */
  ssr: true,
})(MyApp);
```

but was able to resolve it by adding the code below to check for CSR vs. SSR from [Server-Side Rendering (SSR)](https://trpc.io/docs/v9/ssr)
```
    if (typeof window !== 'undefined') {
      // during client requests
      return {
        transformer: superjson, // optional - adds superjson serialization
        url: '/api/trpc',
      };
    }
```

I no longer get CORS errors when fetching from trpc api endpoints from all of the [Vercel deployment domain pattern you mentioned](https://github.com/orgs/vercel/discussions/569).

tangent: also for what its worth, I also tried [enabling CORS for my Next app](https://vercel.com/guides/how-to-enable-cors#enabling-cors-in-a-next.js-app), but I wasn't able to get that to work.
